packages/shared/src/hooks/useFeed.ts

```diff
@@ -185,6 +175,19 @@ export default function useFeed<T>(
               index,
             }),
           );
+          const ad = adsQuery.data?.pages[pageIndex];
```

Comment:
Also have some concerns here.
I'm not quite sure this will work as the ads might be loading or returning 0.

We want to show the placeholder while they are loading (post + ads)

Reply:
But this is how it worked before, I just refactored it. Is there a mistake?

Reply:
This is how it was before:
```
          if (adsQuery.data?.pages[pageIndex]) {	
            posts.splice(adSpot, 0, {	
              type: 'ad',	
              ad: adsQuery.data?.pages[pageIndex],	
            });	
          } else {	
            posts.splice(adSpot, 0, {	
              type: 'placeholder',	
            });
          }
```

And this is how it is now:
```
          const ad = adsQuery.data?.pages[pageIndex];
          posts.splice(
            adSpot,
            0,
            ad
              ? {
                  type: 'ad',
                  ad,
                }
              : {
                  type: 'placeholder',
                },
          );

```

Reply:
NVM reverted